core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceFactory.java

```diff
@@ -166,20 +168,24 @@ public static List<AnnotatedServiceElement> find(
                       .collect(toImmutableList());
     }
 
-    private static HttpStatus defaultResponseStatus(Method method) {
+    private static HttpStatus defaultResponseStatus(Method method, Class<?> clazz) {
         final StatusCode statusCodeAnnotation = AnnotationUtil.findFirst(method, StatusCode.class);
-        if (statusCodeAnnotation == null) {
-            // Set a default HTTP status code for a response depending on the return type of the method.
-            final Class<?> returnType = method.getReturnType();
-            return returnType == Void.class ||
-                   returnType == void.class ||
-                   KotlinUtil.isSuspendingAndReturnTypeUnit(method) ? HttpStatus.NO_CONTENT : HttpStatus.OK;
+        if (statusCodeAnnotation != null) {
+            final int statusCode = statusCodeAnnotation.value();
+            checkArgument(statusCode >= 0,
+                          "invalid HTTP status code: %s (expected: >= 0)", statusCode);
```

Comment:
Not releated to this PR but we can remove this validation because `HttpStatus.valueOf()` throws IAE if the status is negative.
```suggestion
```